cmd/clusterctl/pkg/client/common.go

```diff
@@ -67,9 +67,8 @@ func parseProviderName(provider string) (name string, version string, err error)
 	}
 
 	name = t[0]
-	errs := validation.IsDNS1123Label(name)
-	if len(errs) != 0 {
-		return "", "", errors.Errorf("invalid name value: %s", strings.Join(errs, "; "))
+	if err := validateDNS1123Label(name); err != nil {
+		return "", "", errors.Errorf("invalid provider name %q. Provider name should be in the form name[:version] and the name should be valid", provider)
```

Comment:
This ignores `err` and doesn't return its details?

Reply:
fixed by using `errors.Wrapf`